packages/core/src/wallet/metamaskBridge.ts

```diff
@@ -24,36 +28,34 @@ function isMetaMaskProvider(obj: unknown): obj is MetaMaskProvider {
 function detectMetaMaskProvider(
   windowObject: Record<string, unknown>,
   { timeout = 3000 } = {},
-) {
-  let handled = false
+): Promise<MetaMaskProvider | null> {
   return new Promise<MetaMaskProvider | null>((resolve) => {
-    if (windowObject.ethereum) {
-      handleEthereum()
-    } else {
-      if (typeof windowObject.addEventListener === "function") {
-        windowObject.addEventListener("ethereum#initialized", handleEthereum, {
-          once: true,
-        })
+    let handled = false
+
+    const handleEIP6963Provider = (event: CustomEvent) => {
+      const { provider } = event.detail
+      if (isMetaMaskProvider(provider)) {
```

Comment:
```suggestion
      if (info.rdns === "io.metamask" && isMetaMaskProvider(provider)) {
```

instead of check if a provider isMetamask, as i found that Okx Provider also contains the same value
hence, we may check the rdns name from the event and select the correct provider